website_base_multi_image/static/src/js/website_multi_image.js

```diff
@@ -0,0 +1,96 @@
+/* © 2016 Tecnativa, S.L.
+ * License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html).
+ */
+odoo.define("website_base_multi_image.images",
+function(require){
+"use strict";
+
+var animation = require("web_editor.snippets.animation");
+var core = require("web.core");
+var Model = require("web.Model");
+var Session = require("web.Session");
+var _t = core._t;
+var $ = require("$");
+
+animation.registry.website_base_multi_image =
+animation.Class.extend({
+    selector: ".js_image_container",
+    start: function (editable_mode) {
+        if (editable_mode) {
+            return;
+        };
+        this.$image_preview_thumb = this.$(".thumbnail-images");
+        this.$main_image = this.$("#main_image");
+        this.$file_input = this.$("#file_input");
+        this.$label_input = this.$(".image_upload");
+        this.$file_input_multi = this.$("#file_input_multi");
+        this.image_added_id = 0;
+        core.qweb.add_template(
+            "/website_base_multi_image/static" +
+            "/src/xml/website_multi_image.xml");
+        this.bind_events();
+    },
+    bind_events: function () {
+        var this_ = this;
+        this.$el.on("change", "#file_input", function (event) {
+            return this_.image_preview(event);
+        });
+    },
+    image_preview: function(e){
+        var image = e.target.files[0];
+        var reader = new FileReader();
+        var self = this;
+        reader.onload = (function(file) {
+           return function(e) {
```

Comment:
This is a bit cumbersome, I guess you do it to get the file name after you read its binary data. You can simply [save the name inside `reader`](http://stackoverflow.com/a/33520129/1468388) and bind a proxied method to it:

```
image_preview: function (event) {
    var image = e.target.files[0];
    var reader = new FileReader();
    reader.fileName = image.name;
    reader.onload = $.proxy(this.render_load, this);
},
reader_load: function (event) {
    // Inside here...
    // event.target.name === previous image.name
    // this === self
   ...
}
```